src/AWS.Deploy.Common/Recipes/OptionSettingItem.ValueOverride.cs

```diff
@@ -102,13 +102,16 @@ public async Task SetValue(IOptionSettingHandler optionSettingHandler, object va
                     var result = await validator.Validate(valueOverride);
                     if (!result.IsValid)
                     {
-                        throw new ValidationFailedException(DeployToolErrorCode.OptionSettingItemValueValidationFailed,
-                            result.ValidationFailedMessage?.Trim() ?? $"The value '{valueOverride}' is invalid for option setting '{Name}'.");
+                        Validation.ValidationStatus = ValidationStatus.Invalid;
```

Comment:
The validation object is basically maintaining the state of the last set value attempt. Since we are not blocking the deployment based on the validation object status then we should be fine. Though for the Toolkit, the Validation object can be considered as the source of truth since it will be mirroring what the user has on the UI.